docs/src/utils/demoState.ts

```diff
@@ -0,0 +1,20 @@
+/*
+
+Problem: Component demo state gets wiped out by NextJS when a customer switches themes (e.g., toggles between light/dark mode). 
+- See: https://github.com/aws-amplify/amplify-ui/issues/678
+- This has to do with how we are doing dynamic imports, which give us performance benefits
+
+Solution: We are using a simple Map to persist component demo state across NextJS re-renders without the complexity of state management (e.g., Redux)
+
+We are not using React Context for two reasons (based on the React Context docs: https://reactjs.org/docs/context.html):
+- Since the props for each component's demo do not need to be shared across the entire docs site, 
+  these props aren't "global" state in the same way that a current authenticated user, theme, or preferred language would be
```

Comment:
I think the reason might not matter in our case because we NextJS only render the page user is looking at... 😁